drivers/spi/spi_ll_stm32.c

```diff
@@ -771,11 +746,13 @@ static bool buf_in_nocache(uintptr_t buf, size_t len_bytes)
 	}
 #endif /* CONFIG_NOCACHE_MEMORY */
 
-	for (size_t i = 0; i < ARRAY_SIZE(nocache_mem_regions); i++) {
-		const struct mem_region *mem_reg = &nocache_mem_regions[i];
+	const struct mem_attr_region_t *regions;
+	size_t num_regions =  mem_attr_get_regions(&regions);
+
+	for (size_t i = 0; i < num_regions; i++) {
```

Comment:
Do we really need to iterate across regions ?
From my understanding, `mem_attr_check_buf()` will iterate across the array of regions.

Then, I didn't had coffee so maybe I'm not woken up enough, but if iteration is required, I wonder how it happens in your code

Reply:
Oops, yes, this loop is completely unnecessary. Fixed now, thanks.